lib/brains/tests/base.py

```diff
@@ -1,18 +1,83 @@
+import mock
+
 from lib.brains import client
 from lib.brains.errors import MockError
 from solitude.base import APITest
 
 
+class BraintreeMock(mock.Mock):
+
+    def __getattr__(self, name):
+        """
+        Record that a mock method was registered so we can check it
+        was called.
+        """
+        if not name.startswith('_'):
+            setattr(self, '_registered', [])
+            self._registered.append(name)
+        return super(BraintreeMock, self).__getattr__(name)
+
+    def tolist(self, *args, **kw):
+        """
+        DRF can get into a infinite loop with Mock on this method. If you
+        override the mock so it returns a value it works.
+
+        This is a hack of the worst kind.
+        """
+        raise MockError(
+            'Attempt to serialise a Mock without the result being overridden.')
+
+
 class BraintreeTest(APITest):
+    """
+    A standard APITest with the ability to provide Braintree mocks.
+
+    Add in dict of gateways to override onto the class, for example:
+
+            gateways = {'client': ClientTokenGateway}
+
+    Access the mock lke this:
+
+            self.mocks['client'].generate.return_value = 'a-sample-token'
```

Comment:
I know it's more letters to type but I think this would make more sense as:

    self.mocks['ClientTokenGateway'].generate.return_value

you wouldn't need a mapping like that either